profiles/utils.py

```diff
@@ -0,0 +1,121 @@
+""" Utils for profiles """
+from os import path
+from urllib.parse import urljoin
+
+from contextlib import contextmanager
+from django.conf import settings
+from django.core.files.temp import NamedTemporaryFile
+
+from PIL import Image
+
+from open_discussions.utils import now_in_utc
+
+
+# This is the Django ImageField max path size
+IMAGE_PATH_MAX_LENGTH = 100
+
+# Max dimension of either height or width for small and medium images
+IMAGE_SMALL_MAX_DIMENSION = 64
+IMAGE_MEDIUM_MAX_DIMENSION = 128
+
+IMAGE_PATH_PREFIX = 'profile'
+
+default_profile_image = urljoin(settings.STATIC_URL, "images/avatar_default.png")
+
+
+def image_uri(profile, image_field='image_small'):
+    """ Return the correctly formatted image URI """
+    image_file = getattr(profile, '{}_file'.format(image_field))
+    if not image_file.name:
+        return getattr(profile, image_field) or default_profile_image
+    return image_file.url
+
+
+def generate_filepath(filename, suffix=''):
+    """
+    Generate and return the filepath for an uploaded image
+
+    Args:
+        filename(str): The name of the image file
+        suffix(str): 'small', 'medium', or ''
+
+    Returns:
+        str: The filepath for the uploaded image.
+    """
+    name, ext = path.splitext(filename)
+    timestamp = now_in_utc().replace(microsecond=0)
+    path_format = "{prefix}/{name}-{timestamp}{suffix}{ext}"
```

Comment:
Could we namespace this into a directory of `user.username`?

Reply:
✅